operator-common/src/main/java/io/strimzi/operator/common/operator/resource/DeploymentOperator.java

```diff
@@ -87,4 +89,20 @@ private Deployment incrementGeneration(Deployment deployment) {
 
         return super.internalPatch(namespace, name, current, desired, cascading);
     }
+
+    @Override
+    protected Future<ReconcileResult<Deployment>> internalCreate(String namespace, String name, Deployment desired) {
+        setGeneration(desired, INIT_GENERATION);
+        Future<ReconcileResult<Deployment>> result = Future.future();
+
+        Future<ReconcileResult<Deployment>> dep = super.internalCreate(namespace, name, desired);
+        dep.compose(res -> result.complete(res), result);
+        return result;
```

Comment:
Why do you need `result` here? You can just `return super.internalCreate(namespace, name, desired);`